hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapUnboundReturnValuesTestSupport.java

```diff
@@ -178,6 +179,13 @@ private void fillToUpperLimit(IMap<Integer, Integer> fillMap, IMap<Integer, Inte
         assertEquals("Expected map size of client map to match upperLimit", upperLimit, queryMap.size());
     }
 
+    private void checkException(QueryResultSizeExceededException e) {
```

Comment:
What wat?

This is done to ensure, that we don't execute the query a second time on partitions, after it already failed with `QueryResultSizeExceededException` on members (which may have a serious performance impact). This is one of the bugs which is fixed in this PR.